islandora.module

```diff
@@ -331,6 +333,175 @@ function islandora_form_alter(&$form, FormStateInterface $form_state, $form_id)
       }
     }
   }
+  $form_object = $form_state->getFormObject();
+  $utils = \Drupal::service('islandora.utils');
+
+  if ($form_object instanceof EntityForm) {
+    $entity = $form_object->getEntity();
+
+    if ($entity->getEntityTypeId() == "node" && $utils->isIslandoraType($entity->getEntityTypeId(), $entity->bundle()) && strpos($form['#form_id'], 'delete_form') !== FALSE) {
+      $medias = $utils->getMedia($form_state->getFormObject()->getEntity());
+      if (count($medias) != 0) {
+        $form['delete_associated_content'] = [
+          '#type' => 'checkbox',
+          '#title' => t('Delete all associated medias and nodes'),
+        ];
+
+        $media_list = [];
+
+        foreach ($medias as $media) {
+          $media_list[] = $media->getName();
+        }
+
+        $form['media_items'] = [
+          '#theme' => 'item_list',
+          '#type' => 'ul',
+          '#items' => $media_list,
+          '#attributes' => ['class' => 'mylist'],
+          '#wrapper_attributes' => ['class' => 'container'],
```

Comment:
I don't have definitive documentation on this to hand, but I seem to recall that `class` attributes are expected to be passed as array.

```suggestion
          '#attributes' => ['class' => ['mylist']],
          '#wrapper_attributes' => ['class' => ['container']],
```

That said, I am not sure if these attributes are entirely appropriate:
* `mylist` might be too generic. Having the class contain some context about what _class_ of element it represents would probably be useful, but the concept of it being a list is already contained in the HTML element that is to be rendered, so, something like `islandora-media-items`?
* `container`... I'm not sure there _is_ a wrapper element by default for an `item_list`, so it might not have any effect? Might instead want to add a `'#theme_wrapper' => ['container']` kind of thing, which I believe would include the `container` class (on a `div` element which is wrapped around the list).